src/MUSI6106Exec/MUSI6106Exec.cpp

```diff
@@ -1,131 +1,311 @@
+                }
+                else if (!strcmp(argv[i], "freq")) {
+                    result.convolver = CFastConv::ConvCompMode_t::kFreqDomain;
+                }
+                else {
+                    throw(std::invalid_argument("unknown argument!"));
+                }
+            }
+            else if (argv[i][1] == 'g') {
+                if (i == argc - 1) throw(std::invalid_argument("Incomplete argument of wet gain!"));
+                result.wetGain = std::stof(argv[++i]);
+            }
+            else if (argv[i][1] == 'b') {
+                if (i == argc - 1) throw(std::invalid_argument("Incomplete argument of block size!"));
+                result.blockSize = std::stoi(argv[++i]);
+            }
+            else if (argv[i][1] == 'i') {
+                if (i == argc - 1) throw(std::invalid_argument("Incomplete argument of input file path!"));
+                result.inputPath = std::string(argv[++i]);
+            }
+            else if (argv[i][1] == 'r') {
+                if (i == argc - 1) throw(std::invalid_argument("Incomplete argument of IR file path!"));
+                result.IRPath = std::string(argv[++i]);
+            }
+            else if (argv[i][1] == 'o') {
+                if (i == argc - 1) throw(std::invalid_argument("Incomplete argument of output file path!"));
+                result.outputPath = std::string(argv[++i]);
+            }
+            else if (argv[i][1] == 'h' || !strcmp(argv[i], "--help")) {
+                printUsage();
+                exit(0);
+            }
+            else {
+                throw(std::invalid_argument("unknown argument!"));
+            }
+        }
+    }
+    return result;
+}
 
-    static const int            kBlockSize = 1024;
-    long long                   iNumFrames = kBlockSize;
-    int                         iNumChannels;
 
-    float                       fModFrequencyInHz;
-    float                       fModWidthInSec;
+/*
+    This class hides unimportant low-level details like:
+        - Two phase initialization / deletion of `CAudioFileIf`
+        - Nested initialization / deletion of audio buffer
+        - Manage audio file related information (IO type, file spec)
+        - Error handling
+
+    This class also provides easier interface:
+        - Only need `filepath`, `blockSize`, and `ioType` to create.
+
+    This class represents the true "conceptual" audio file, which is easier to understand & use.
+*/
+class AudioFileWrapper {
+public:
+    AudioFileWrapper(
+        const std::string& filepath, 
+        int blockSize, 
+        CAudioFileIf::FileIoType_t ioType, 
+        CAudioFileIf::FileSpec_t* fileSpec = nullptr
+    ): filepath(filepath), blockSize(blockSize), ioType(ioType), fileSpec(fileSpec) {
+        CAudioFileIf::create(phAudioFile);
+
+        if (phAudioFile == nullptr) {
+            // throw std::exception("failed to create audio file handler");
```

Comment:
It's because `std::exception(const& std::string)` seems to be MSFT's c++ compiler's feature, and this will fail to be compiled on other platforms. So it was commented out. Maybe I should give it more meaningful exceptions like `std::invalid_argument()` in simple parser.

Reply:
Fixed: 95304ecc833c48f2f506afc968ddf77fa9f00681.